source/extensions/access_loggers/http_grpc/grpc_access_log_impl.h

```diff
@@ -125,13 +126,12 @@ class HttpGrpcAccessLog : public AccessLog::Instance {
       envoy::data::accesslog::v2::AccessLogCommon& common_access_log,
       const StreamInfo::StreamInfo& stream_info);
 
-  // AccessLog::Instance
-  void log(const Http::HeaderMap* request_headers, const Http::HeaderMap* response_headers,
-           const Http::HeaderMap* response_trailers,
-           const StreamInfo::StreamInfo& stream_info) override;
-
 private:
-  AccessLog::FilterPtr filter_;
+  // Common::ImplBase
+  void emitLog(const Http::HeaderMap& request_headers, const Http::HeaderMap& response_headers,
+               const Http::HeaderMap& response_trailers,
+               const StreamInfo::StreamInfo& stream_info) override;
```

Comment:
why private? usually override methods should be public or protected.

Reply:
The public interface AccessLog::Instance::log(), which will this overridden emitLog() function.